zerver/views/events_register.py

```diff
@@ -114,10 +125,11 @@ def events_register_backend(
         event_types,
         queue_lifespan_secs,
         all_public_streams,
-        narrow=narrow,
-        include_subscribers=include_subscribers,
-        client_capabilities=client_capabilities,
-        fetch_event_types=fetch_event_types,
-        spectator_requested_language=spectator_requested_language,
+        include_subscribers,
+        include_streams,
+        client_capabilities,
+        narrow,
+        fetch_event_types,
+        spectator_requested_language,
```

Comment:
Why the change in calling conventions for these? 

Generally we don't want functions with very large numbers of positional arguments. @PIG208 @andersk do you think we should create a `dataclass` for the collection of parameters passed to `do_events_register`? We can then rebase this PR on top of that.

Reply:
I think it's worth noting that inside `do_events_register` there are also functions like this, such as `request_event_queue` and `fetch_initial_state_data`. We might want to identify some parameters that can be grouped together and define data structures representing them through composition.

Reply:
Either `TypedDict` or `dataclass` should be fine. Probably `dataclass` is preferable since we will need default values.

Reply:
I don’t see any benefit of building a structure type for “the collection of parameters passed to `do_events_register`”. All we’re going to do with that structure is put a bunch of parameters into it and then immediately pull them back out again. Keyword arguments accomplish the same goal with strictly less syntactic noise.

If we want to prevent some or all of these arguments from being passed positionally, we can mark them keyword-only.

Reply:
Yeah, I think I agree keyword-only would probably be fine here, as long as we're not passing the bundle around to other functions that `do_events_register` calls, etc. In any case, @amanagr can we undo the change here switching to positional parameters?